tests/unit/remote_build/test_repo.py

```diff
@@ -108,6 +108,15 @@ def test_repo_push_remote_error(self):
         )
         self.assertThat(str(raised), Contains("is not supported"))
 
+    def test_branch_name_detached_head(self):
+        repo = Repo(self._dir.path)
+        repo.add("foo")
+        repo.commit()
+        commit = self._check_output("git", "rev-parse", "HEAD")
+        commit = commit.decode("ascii").strip()
```

Comment:
random question, why decode to ascii?

Reply:
It could have been utf-8, but since it's only a commit hash ascii will do. The decoding itself was used to get a string from bytes.

Reply:
yeah, the question was, why ascii :-)

Reply:
It could be nothing and use the default, I can change it if you think it's neater.